lib/std/mem.zig

```diff
@@ -3427,20 +3427,57 @@ pub fn swap(comptime T: type, a: *T, b: *T) void {
     b.* = tmp;
 }
 
+inline fn reverseVector(comptime N: usize, comptime T: type, a: []T) [N]T {
+    var res: [N]T = undefined;
+    inline for (0..N) |i| {
```

Comment:
The `inline` doesnt affect the generated assembly so it is unnecessary here. From the zig docs:

> It is recommended to use inline loops only for one of these reasons:
> * You need the loop to execute at comptime for the semantics to work.
> * You have a benchmark to prove that forcibly unrolling the loop in this way is measurably faster.

the same thing goes for the `inline fn`:

> It is generally better to let the compiler decide when to inline a function, except for these scenarios:
> * To change how many stack frames are in the call stack, for debugging purposes.
> * To force comptime-ness of the arguments to propagate to the return value of the function, as in the above example.
> * Real world performance measurements demand it.

Reply:
thanks! ill update it

Reply:
This is untrue. There should never be a call to the reverseVector function. 

Similarly, the inline for loop is here to ensure unrolling of the loop.

Reply:
upon closer inspection the inlines are both actually necessary for reasonable codegen in releasesmall. in releasefast its fine without them however.
below are all in releasesmall
[both inlines](https://godbolt.org/z/fY1erKjc7)
[only on function](https://godbolt.org/z/99cMqob9v)
[only on loop](https://godbolt.org/z/frGThK7bn)

Reply:
> This is untrue. There should never be a call to the reverseVector function.
> 
> Similarly, the inline for loop is here to ensure unrolling of the loop.

good catch! i shouldve verified it myself before making code changes.